### DISCUSSION TITLE
element.updated hook doesn't give the updated html

### DISCUSSION THREAD

**Discussion by jochemgruter** [livewire/livewire#4364] [1. Help]
With the `element.updated` hook I try to get the new updated html but it always gives the orignal html.

```
Livewire.hook('element.updated', (el, component) => {
   console.log(el.outerHTML);
}
```
Is there any way to get the updated html of the element?

I need this because an alpine component breaks down when the element is updated. This is the whole code

```
init: function(){
    this.content = this.$refs.content.textContent.trim();

    Livewire.hook('element.updated', (el, component) => {
        if(el === this.$el){
            this.content = // here I need to update the new updated content from livewire
        }
    })
},
```

**Comment by denizozturk** [comment]
Hi, you can have the livewire component trigger a browser event in the update method in the class file.

for example 
`$this->dispatchBrowserEvent('getComponentId')`

JS event:
```
window.addEventListener('getComponentId', event => {
	console.log(event)
})
```

I hope that the information in your work in `event.target` will be useful.

**Comment by joshhanley** [comment]
@jochemgruter yep that will be because the element changes have been made, but the browser won't have done a render yet, so you won't get the updated html until after that render is done.

But I'd be looking at the Alpine component first and so we can see if there is another way around it. Can you share it and what is currently happening and what should be happening?

Hope this helps!
